components/UserObserver.php

```diff
@@ -67,31 +74,129 @@ public function updateUser($params)
+					throw new Exception("Unable to save user contact: ".$username);
 				}
-			} catch (Exception $e) {
-				// silently return back to UserIdentity without having refreshed the user
-			}
+			/*} catch (Exception $e) {
+				\Yii::log("Unable to update user. Error: ". $e->getMessage(), \CLogger::LEVEL_ERROR);
+			}*/
+		//}
+	}
+
+
+	/**
+     * Finds User by username
+     *
+     * @param string $username
+     * @return User
+     */
+	private function getUser(string $username): User
+	{
+		$criteria = new \CDbCriteria();
+		$criteria->join = 'JOIN user_authentication ua ON t.id = ua.user_id';
+		$criteria->addCondition('ua.username = :username');
+		$criteria->params[':username'] = $username;
+		$user = \User::model()->find($criteria);
+		return $user;
+	}
+
+	/**
+     * Finds UserAuthentication by username
+     *
+     * @param string $username
+     * @return UserAuthentication
+     */
+	private function getUserAuthentication(string $username): UserAuthentication
+	{
+		$criteria = new \CDbCriteria();
+		$criteria->addCondition('username = :username');
+		$criteria->params[':username'] = "WILLIAMSS";//$params['username'];
+		$user_authentication = \UserAuthentication::model()->find($criteria);
+		return $user_authentication;
+	}
+
+	/**
+     * Saves new User data which is coming from $remote_user
+     *
+     * @param User $user
+	 * @param UserAuthentication $user_authentication
+	 * @param array $remote_user
+     * @return User
+     */
+	private function saveUser(User $user, UserAuthentication $user_authentication, array $remote_user): User
+	{
+		$user->code = $remote_user['code'];
+		$user_authentication->username = $remote_user['username'];
+		$user->first_name = $remote_user['first_name'];
+		$user->last_name = $remote_user['last_name'];
+		$mehstaffdb_default_email = Yii::app()->params['mehstaffdb_default_email'];
+		if(strlen($mehstaffdb_default_email) != 0) {
+			$user->email = $mehstaffdb_default_email;
+		}
+		$user->title = $remote_user['title'];
+		$user->qualifications = $remote_user['qualifications'];
+		$user->role = $remote_user['role'];
+		$user->doctor_grade_id = $this->getDoctorGradeFromJobTitle($remote_user['role']);
+		if(isset($remote_user['registration_code']) && isset($remote_user['registration_code'][0]['ProfessionalRegistration'])) {
+			$user->registration_code = $this->getGMCRegistrationNumber($remote_user['registration_code'][0]['ProfessionalRegistration']);
+		}
+		//$user->password = 'faed6633f5a86241f3e0c2bb2bb768fd';
+		$user->is_consultant = $remote_user['is_consultant'];
+		$user->is_surgeon = $remote_user['is_surgeon'];
+		$user_authentication->active = !$remote_user['active'];
+		$user->global_firm_rights = 1;
+
+		if (!$user->save(false)) {
+			\Yii::log('Unable to save user: '.print_r($user->getErrors(),true), \CLogger::LEVEL_ERROR);
+			throw new Exception('Unable to save user: '.print_r($user->getErrors(),true));
 		}
+
```

Comment:
does user_authentication need to be saved as well?

Reply:
Yes, it does. Now it will save user_authentication too.